packages/workshop-app/app/utils/mdx.tsx

```diff
@@ -6,6 +7,35 @@ import { LaunchEditor } from '~/routes/launch-editor'
 import { AnchorOrLink } from './misc'
 import Accordion from '~/components/accordion'
 
+function getCode(data: any) {
+	// just in case we are lost in space
+	try {
+		if (typeof data === 'string') return data
+		const { children } = data.props
+		if (typeof children === 'string') return children
+		return children.map(getCode).flat().join('')
+	} catch {}
+	return null
+}
+
+export function preWithCopyToClipboard({ children, ...props }: any) {
+	const showCopyButton = !Object.keys(props).find(att => att === 'data-nocopy')
+	const codeToCopy = showCopyButton && getCode(children)
+	return (
+		<div className="relative">
+			{codeToCopy ? (
+				<button
```

Comment:
Let's make it so this button has opacity 0 unless the parent `div` is hovered or if it is focused.

Reply:
done in 863ac5d, nut now keyboard accessible when the button is not visible